### PR TITLE
adding SUBSTRING function

### PR DESCRIPTION
**Pull request by rachithrr**

### CODE REVIEW

File: sql-preview/functions/_posts/2022-12-09-sql-reverse.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ reverse(expr)
 
 #### Arguments
 
-The argument `expr` can either be a string column or another another string function
+The argument `expr` can either be a string column or another string function
```
Comment (pokeeffe-molecula) [inline sql-preview/functions/_posts/2022-12-09-sql-reverse.md:24]: I would reword this even more generally. Something like "the argument `expr` is any expression of type `string`", and be inclined to use this everywhere we discuss parameters.

File: sql-preview/functions/_posts/2022-12-09-sql-substring.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+substring(expr,startIndex,length)
+```
+
+#### Arguments
+
+_expr_ 
+The input string from which to extract the substring. Expr can either be a string column or another string function.
+
+_startIndex_
+The starting index of the substring in the input string, starting at zero. startIndex must be an integer.
```
Comment (pokeeffe-molecula) [inline sql-preview/functions/_posts/2022-12-09-sql-substring.md:28]: Same thing here - let's talk types i.e. "startIndex can be any expression of of type `int`"

File: sql-preview/functions/_posts/2022-12-09-sql-substring.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+_length_ *(optional)*
+The length of the substring to extract. length must be a positive integer
```
Comment (pokeeffe-molecula) [inline sql-preview/functions/_posts/2022-12-09-sql-substring.md:31]: And here - let's talk types i.e. "length is any expression of of type `int`"

**Review by pokeeffe-molecula — CHANGES_REQUESTED** [review]
Few comments in describing parameters

**Review by gthrone25 — COMMENTED** [review]
@rachithrr Looks good! Just a couple of comments to address. It looks like the structural piece of this was already merged in last week, so no turning back now :). @ljsinclair will most likely restructure in the near future.

**Review by ljsinclair — CHANGES_REQUESTED** [review]
Hi there,
Can you also drop the heading levels by 1 please? Start the page at h2 (##) and subsequent headings are h3 (###)

**Review by gthrone25 — APPROVED** [review]
Looks good! Approving and will merge once @pokeeffe-molecula and @ljsinclair approve. Note might be better just to "comment" (as opposed to request change) on reviews in the future otherwise we need all 3 approvals before merging.

**Review by pokeeffe-molecula — APPROVED** [review]
lgtm

**Review by ljsinclair — APPROVED** [review]
super :) Thanks
